tutorials/run-spring-petclinic-on-app-engine-cloudsql.md

```diff
@@ -0,0 +1,131 @@
+```
+   gcloud init
+   gcloud app create
+   gcloud auth application-default login
+```
+1. Clone and test the Pet Clinic application locally:
+```
+    git clone https://github.com/spring-projects/spring-petclinic.git
+    cd spring-petclinic
+    ./mvnw spring-boot:run
+```
+
+  Access http://localhost:8080 via your web browser to see the application homepage.
+
+  By default, the application uses an in-memory [HyperSQL database](http://hsqldb.org/).
+  You will now switch from HyperSQL to using Cloud SQL as your database.
+
+### Using Cloud SQL as your database
+
+1. Enable the [Cloud SQL API](https://console.cloud.google.com/flows/enableapi?apiid=sqladmin).
+
+1. Create a Second Generation Cloud SQL (MySQL) instance and set the root user password following [these instructions](https://cloud.google.com/sql/docs/mysql/create-instance#create-2nd-gen).
+
+1. Create the `petclinic` database.
+```
+    gcloud beta sql databases create petclinic --instance=INSTANCE_NAME
+```
+1. Get the `connectionName` of the instance in the format `project-id:zone-id:instance-id` by running the following command:
+```
+   gcloud beta sql instances describe INSTANCE_NAME
+```
+1. Update `src/main/resources/application-mysql.properties`, replacing INSTANCE_CONNECTION_NAME with the `connectionName` from the previous step:
+```
+    database=mysql
+    spring.datasource.driverClassName=com.mysql.jdbc.Driver
+    spring.datasource.url=jdbc:mysql://google/petclinic?Cloud SQLInstance=INSTANCE_CONNECTION_NAME&socketFactory=com.google.cloud.sql.mysql.SocketFactory
+    spring.datasource.username=root
+    spring.datasource.password=my-smart-password
+```
+1. Update `pom.xml` to include the [Cloud SQL MySQL Socket Factory socket library](https://github.com/GoogleCloudPlatform/cloud-sql-mysql-socket-factory):
+```
+    <dependency>
+        <groupId>com.google.cloud.sql</groupId>
+        <artifactId>mysql-socket-factory</artifactId>
+        <version>1.0.2</version>
+    </dependency>
+```
+   The socket library allows you to connect to your Cloud SQL instance for local testing and deployment.
+1. Restart the Spring Boot application using the mysql [profile](http://docs.spring.io/spring-boot/docs/current/maven-plugin/examples/run-profiles.html):
+```
+   ./mvnw -Drun.profiles=mysql spring-boot:run
+```
+  Access the application homepage http://localhost:8080 via your web browser and add some data.
+1. You can verify the data exists in Cloud SQL by running queries agains the `petclinic` database using the [Cloud Shell](https://cloud.google.com/sql/docs/mysql/quickstart#connect_to_your_instance_using_the_db_client_client_in_the_cloud_shell).
+
+
+### Deploying to App Engine flexible environment on GCP
+Now that you've tested the application locally, you can deploy the application to the App Engine flexible environment.
+Once deployed, it will be accessible at https://<your-project-id>.appspot.com.
```

Comment:
`<your-project-id>` is being stripped out as an HTML-like tag. Either change to `&lt;` or use a different separator.